src/core/js/models/questionModel.js

```diff
@@ -40,55 +40,30 @@ define([
         setupButtonSettings: function() {
             var globalButtons = Adapt.course.get("_buttons");
 
-            // Checks if local _buttons exists and if not use global
+            // Check if  '_buttons' attribute exists and if not use the globally defined buttons.
             if (!this.has("_buttons")) {
                 this.set("_buttons", globalButtons);
             } else {
-                // check all the components buttons
-                // if they are empty use the global default
+                // Check all the components buttons.
+                // If they are empty use the global defaults.
                 var componentButtons = this.get("_buttons");
 
-                if (typeof componentButtons.submit == 'undefined') {
-                    for (var key in componentButtons) {
-                        if (typeof componentButtons[key] == 'object') {
-                          // ARIA labels
-                          if (!componentButtons[key].buttonText && globalButtons[key].buttonText) {
+                for (var key in componentButtons) {
```

Comment:
just a recommendation and not a requirement... but would be a good opportunity to use `const` here instead of `var`